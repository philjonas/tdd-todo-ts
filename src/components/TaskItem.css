.task-item{
    display: flex;
    padding: 10px 5px;
}

.task-item-complete{
    border: 1px black solid;
    margin: 0 5px 0 5px;
    user-select: none;
    width: 15%;
}

.task-item-text{
    width: 70%;
    text-align: start;
    padding: 0 10px;
}

.task-item-delete{
    width: 15%;
}

.completed{
    text-decoration: line-through;
}