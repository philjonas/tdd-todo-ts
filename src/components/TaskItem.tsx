import React from "react";
import "./TaskItem.css";

export interface TaskItemProps {
  id: number;
  value: string;
  isCompleted: boolean;
  onDelete?: (id: number) => void;
  onComplete?: (id: number) => void;
}

export const TaskItem = ({
  id,
  value,
  isCompleted,
  onDelete,
  onComplete,
}: TaskItemProps): JSX.Element => {
  const textClass = isCompleted ? "task-item-text completed" : "task-item-text";
  return (
    <div className="task-item">
      <div data-testid="task-item-text" className={textClass}>
        {value}
      </div>
      <span
        onClick={() => onComplete?.(id)}
        data-testid="task-item-complete"
        className="task-item-complete"
      >
        <label>Completed:</label>
        <input
          data-testid="task-item-completed"
          type="checkbox"
          checked={isCompleted}
          readOnly
        />
      </span>
      <button
        data-testid="task-item-delete"
        onClick={() => onDelete?.(id)}
        className="task-item-delete"
      >
        Delete
      </button>
    </div>
  );
};
