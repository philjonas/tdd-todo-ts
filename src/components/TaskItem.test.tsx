import { render } from "@testing-library/react";
import { TaskItem, TaskItemProps } from "./TaskItem";

const setup = (props: TaskItemProps) => {
  const utils = render(<TaskItem {...props} />);
  const input = utils.getByTestId("task-item-text");
  const completed = utils.getByTestId("task-item-completed") as HTMLInputElement;
  return {
    input,
    completed,
    ...utils,
  };
};

const props: TaskItemProps = {
  id: 1,
  value: "new task",
  isCompleted: false,
  onDelete: (id:number) => jest.fn(),
  onComplete: (id:number) => jest.fn(),
};

describe("TaskItem", () => {
  it("renders correctly", () => {
    const wrapper = <TaskItem {...props} />;
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the value from props", () => {
    const { input } = setup(props);
    expect(input.innerHTML).toBe("new task");
  });

  it("should not be completed on creation", () => {
    const { completed } = setup(props);
    expect(completed.checked).toBe(false);
  });

  it("should be completed from props", () => {
    const { completed } = setup({...props, isCompleted: true});
    expect(completed.checked).toBe(true);
  });

});
