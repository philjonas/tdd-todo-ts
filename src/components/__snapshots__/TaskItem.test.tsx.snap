// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem renders correctly 1`] = `
<TaskItem
  id={1}
  isCompleted={false}
  onComplete={[Function]}
  onDelete={[Function]}
  value="new task"
/>
`;
